src/test/ui/consts/const-eval/promoted_errors.rs

```diff
@@ -8,21 +8,34 @@
 
 #![warn(const_err, arithmetic_overflow, unconditional_panic)]
 
+// The only way to have promoteds that fail is in `const fn` called from `const`/`static`.
+const fn overflow() -> u32 {
+    0 - 1
```

Comment:
@oli-obk unlike in previous versions of this PR, the "arithmetic overflow" and "unconditional panic" lint does not fire here any more. Any idea why that is?

Reply:
Ah, I guess this is because we now have separate CTFE MIR, and do not run const-prop any more on the CTFE MIR. The `fn` is private and otherwise unused so regular MIR is never created.